postRoute/router.js

```
@@ -0,0 +1,58 @@
+const express = require("express");
+
+const router = express.Router();
+const db = require("./postDb");
+
+router.get("/", async (req, res) => {
+  try {
+    const posts = await db.get();
+    res.status(200).json(posts);
+  } catch (error) {
+    res.status(500).json({ message: "Error" });
+  }
+});
+
+router.get("/:id", async (req, res) => {
+  const { id } = req.params;
+  try {
+    const post = await db.getById(id);
+    res.status(201).json(post);
+  } catch (error) {
+    res.status(500).json({ message: "error" });
+  }
+});
+
+router.post("/", async (req, res) => {
+  const post = req.body;
+  try {
```

before even getting into your try/catch block, you should be checking to see if the body has a `text` and `user_id` property, otherwise it should throw a 404 error and not even attempt the db.insert().

that's the first level of authentication this endpoint needs to run, here's a trickier one that you need to think about--making sure that the `user_id` references an  existing user! this would require adding a findById call here to the _users_ database, but it is needed since otherwise you could be adding posts that aren't tethered to any existing user